ingestion/src/metadata/ingestion/source/pipeline/airflow/metadata.py

```diff
@@ -311,8 +332,28 @@ def _build_dag(data: Any) -> SerializedDAG:
 
         return SerializedDAG.from_json(data)
 
+    def get_user_details(self, email) -> Optional[EntityReference]:
+        user = self.metadata.get_user_by_email(email=email)
+        return user
+
+    def get_owners(self, owners) -> Optional[EntityReference]:
+        try:
+            if isinstance(owners, str) and owners:
+                return self.get_user_details(email=owners)
```

Comment:
Here we are not using the right logic. `get_user_details` returns a `User`, not an `EntityReference` (we'll need to fix the hint above).

Your `get_user_details` function should directly return the `EntityReference`:

```python
def get_user_details(self, email) -> Optional[EntityReference]:
    user = self.metadata.get_user_by_email(email=email)
    if user:
        return EntityReference(id=user.id.__root__, type="user")
    return None
```

Then we can use this in both branches of the `if`